### PR TITLE
improve error handling in cloud node controller

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Minor improvements to error handling + logging in cloud node controller

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by yastij — COMMENTED** [review]
/lgtm
/hold

feel free to unhold

File: pkg/controller/cloud/node_controller.go [hunk 1]
```diff
@@ -373,7 +373,7 @@ func getNodeAddressesByProviderIDOrName(ctx context.Context, instances cloudprov
 		providerIDErr := err
 		nodeAddresses, err = instances.NodeAddresses(ctx, types.NodeName(node.Name))
 		if err != nil {
-			return nil, fmt.Errorf("NodeAddress: Error fetching by providerID: %v Error fetching by NodeName: %v", providerIDErr, err)
+			return nil, fmt.Errorf("Error fetching node by providerID: %v, and by NodeName: %v", providerIDErr, err)
```
Comment (cheftako) [inline pkg/controller/cloud/node_controller.go]: We should not be beginning error messages with a capital letter. Also we have labelled the second parameter as NodeName but it appears to be an error.

Comment (cheftako) [reply pkg/controller/cloud/node_controller.go]: In fact the first parameter is labelled as providerID but appears to also be an error.

Comment (andrewsykim) [reply pkg/controller/cloud/node_controller.go]: It's `providerIDErr`, but agreed about not starting errors with capaitals, will fix that shortly 

/hold

Comment (andrewsykim) [reply pkg/controller/cloud/node_controller.go]: the two parameters are both errors, the first for attempting to get node addresseses by provider ID, the second by node name. 

/hold cancel

Comment (cheftako) [reply pkg/controller/cloud/node_controller.go]: maybe make it "and error be NodeName: " just to prevent any confusion.

Comment (andrewsykim) [reply pkg/controller/cloud/node_controller.go]: done

**Pull request merged**